shepherd_bms/src/analyzer.cpp

```diff
@@ -274,4 +273,50 @@ void Analyzer::calcStateOfCharge()
         float distance_from_higher = (bmsdata->min_voltage.val) - ((index / 10) + 2.9);
         bmsdata->soc = ((distance_from_higher*STATE_OF_CHARGE_CURVE[index+1]) + ((1-distance_from_higher)*STATE_OF_CHARGE_CURVE[index]));
     }
+}
+
+uint8_t Analyzer::calcThermStandardDev()
+{
+    uint16_t sum_diff_sqrd = 0;
+    for(uint8_t chip = 0; chip < NUM_CHIPS; chip++)
+    {
+        for(uint8_t therm = 17; therm < 28; therm++)
+        {
+            uint16_t sum_diff =  abs(bmsdata->chip_data[chip].thermistor_value[therm] - bmsdata->avg_temp);
+            sum_diff_sqrd += sum_diff * sum_diff;
+        }
+    }
+
+    uint8_t standard_dev = sqrt(sum_diff_sqrd / 88);
+    if(standard_dev < 8)
```

Comment:
What does this do? Is this just to assert a minimum deviation of 8C?

Reply:
Yep, since thats a pretty normal range and we don't wanna ignore anything if its within expected values, we only wanna ignore noisy things when the standard deviation is larger.